### PR TITLE
New promobox styles behind a flag

### PR DESCRIPTION
**Pull request by andygnewman**
@i-like-robots - can you take a look at this..

New style of promo boxes;

screen shot 2015-07-03 at 15 38 17
screen shot 2015-07-03 at 15 38 28
screen shot 2015-07-03 at 15 39 09

Have created separate js and scss files for new vs existing.

Behind flag newPromoBoxStyling on toggler (defaulted to off)

### CODE REVIEW

File: client/components/article/_promo-box-new.scss [hunk 1]
```diff
@@ -0,0 +1,113 @@
+.promo-box {
+	margin-top: 25px;
+	padding: 10px;
+	background-color: oColorsGetPaletteColor('warm-2');
+	> * {
+		margin: 10px 0 0;
+	}
+	> *:first-child {
+		margin-top: 0px;
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: No unit required

File: client/components/article/_promo-box-new.scss [hunk 2]
```diff
@@ -0,0 +1,113 @@
+.promo-box {
+	margin-top: 25px;
+	padding: 10px;
+	background-color: oColorsGetPaletteColor('warm-2');
+	> * {
+		margin: 10px 0 0;
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: This rule seems to get overwritten a lot, maybe only apply this to specific elements instead?

File: client/components/article/_promo-box-new.scss [hunk 3]
```diff
@@ -0,0 +1,113 @@
...
+	> .promo-box__title__wrapper {
+		margin-top: -45px;
+		margin-left: -10px;
+	}
+	> .o-expander__toggle {
+		@include oColorsFor(expander-text, text);
+		margin: 0;
+		padding: 0;
+		background: none;
+		font-family: oFontsGetFontFamilyWithFallbacks(BentonSans);
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: We don't have BentonSans, take a look at https://next.ft.com/__styleguide/design-primitives

File: client/components/article/_promo-box-new.scss [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+		@include nLinksHeadline();
+		border-bottom: 0;
+	}
+}
+
+.promo-box__long__image {
+	@include oGridRespondTo(S) {
+		float: left;
+		margin-right: 10px;
+		width: inherit;
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: inherit? Maybe add a comment what it's inheriting.

File: client/components/article/_promo-box.scss [hunk 5]
```diff
@@ -0,0 +1,46 @@
...
+}
+.article__promo-box__title {
+	font-size: 16px;
+	font-weight: bold;
+	line-height: 1em;
+	text-transform: uppercase;
+}
+.article__promo-box__headline {
+	font-size: 25px;
+	line-height: 1em;
```
Comment (i-like-robots) [inline client/components/article/_promo-box.scss]: No need for the unit in this case, `1` is fine.

**Comment by andygnewman** [comment]
@i-like-robots thanks for the feedback - amends made, hopefully we can ship on Monday morning

File: client/components/article/_promo-box-new.scss [hunk 6]
```diff
@@ -0,0 +1,104 @@
...
+			@include oColorsFor(expander-icon, text);
+			display: inline-block;
+			padding: 0 6px;
+			box-sizing: border-box;
+		}
+		&:hover {
+			text-decoration: none;
+		}
+	}
+	> .o-expander__toggle[aria-pressed] i {
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: these declarations can live inside the one above

File: client/components/article/_promo-box-new.scss [hunk 7]
```diff
@@ -0,0 +1,104 @@
...
+	&:after {
+		content: '';
+		display: table;
+		clear: both;
+	}
+}
+
+.promo-box__long {
+	float: none;
+	max-width: 100%;
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: Does this do anything?

File: client/components/article/_promo-box-new.scss [hunk 8]
```diff
@@ -0,0 +1,104 @@
...
+	}
+
+	&:after {
+		content: '';
+		display: table;
+		clear: both;
+	}
+}
+
+.promo-box__long {
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: `--`

File: client/components/article/_promo-box-new.scss [hunk 9]
```diff
@@ -0,0 +1,104 @@
+.promo-box {
+	margin-top: 25px;
+	padding: 10px;
+	background-color: oColorsGetPaletteColor('warm-2');
+	> *:first-child {
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: Is this necessary? If so `*` is redundant.

File: client/components/article/_promo-box-new.scss [hunk 10]
```diff
@@ -0,0 +1,104 @@
...
+	}
+}
+.promo-box__content {
+	display: inline;
+}
+.promo-box__content p {
+	margin: 5px 0 0;
+	font-size: 18px;
+	line-height: 24px;
+	font-weight: normal;
```
Comment (i-like-robots) [inline client/components/article/_promo-box-new.scss]: Can we use `nType` mixins?

File: client/components/article/_promo-box.scss [hunk 11]
```diff
@@ -0,0 +1,46 @@
...
+	}
+}
+.article__promo-box__image {
+	width: 100%;
+}
+.article__promo-box__content p {
+	margin: 5px 0 0;
+	font-size: 18px;
+	line-height: 24px;
+	font-weight: normal;
```
Comment (i-like-robots) [inline client/components/article/_promo-box.scss]: font mixin?

File: server/transforms/promo-box-new.js [hunk 12]
```diff
@@ -0,0 +1,107 @@
...
+		var promoContentExpansion = $promoBoxIntro.children().length > 2;
+		var promoBoxLong = ($promoBoxIntro.length &&
+												(($promoBoxIntro.html().split(' ').length > 35 &&
+												$promoBoxImage.length) ||
+												$promoBoxIntro.html().split(' ').length > 80 ));
+		var $promoBoxIntroInitial = $('<div></div>')
+			.addClass('promo-box__content__initial')
+			.html($promoBoxIntro.children('p').filter(function(index, el) {
+				if (index < 2) { return $(this); }
+				})
```
Comment (i-like-robots) [inline server/transforms/promo-box-new.js]: funky formatting

File: server/transforms/promo-box-new.js [hunk 13]
```diff
@@ -0,0 +1,107 @@
...
+			);
+		var $promoBoxIntroExtension = $('<div></div>')
+			.addClass('promo-box__content__extension')
+			.html($promoBoxIntro.children('p').filter(function(index, el) {
+				if (index >= 2) { return el; }
+				})
```
Comment (i-like-robots) [inline server/transforms/promo-box-new.js]: as above

File: server/transforms/promo-box-new.js [hunk 14]
```diff
@@ -0,0 +1,107 @@
...
+							$('<h3></h3>')
+								.addClass('promo-box__title')
+								.html($promoBoxTitleChild.length ? $promoBoxTitleChild.html() : $promoBoxTitle.html())
+						)
+			);
+		} else {
+			$promoBox.append(
+				$('<div></div>')
+					.addClass('promo-box__title__wrapper')
+						.append(
```
Comment (i-like-robots) [inline server/transforms/promo-box-new.js]: no need to nest, same collection

File: test/server/transforms/promo-box-new.test.js [hunk 15]
```diff
@@ -0,0 +1,172 @@
...
+				'<div class="promo-box__content o-expander__content">' +
+				'<div class="promo-box__content__initial"><p><strong>Breakthrough:</strong> &#x201C;Closing our first seed round in 10 days three times oversubscribed gave us momentum [to carry through] to the execution of our strategy and into the IPO.&#x201D;</p>' +
+				'<p><strong>Best mentor:</strong> &#x201C;Chris Baohm, my boss at Gresham Partners in Australia&#x2009;.&#x2009;.&#x2009;.&#x2009;made me understand the importance of breaking down complex situations into the core commercial objectives.&#x201D;</p></div>' +
+				'<div class="promo-box__content__extension"><p><strong>Biggest mistake:</strong> &#x201C;It became very clear that we needed a strong team with us who we could trust to navigate the huge due diligence tasks we had in several countries, in a different language.&#x201D;</p>' +
+				'<p><strong>Networking:</strong> &#x201C;Wherever possible, and especially in Latin America, meet in person. Regardless of how good tech is, a face-to-face meeting is irreplaceable.&#x201D;</p></div></div>' +
+				'<button class="o-expander__toggle o--if-js"></button>' +
```
Comment (i-like-robots) [inline test/server/transforms/promo-box-new.test.js]: content?

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-07-07/08:32-2498/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-07-07/08:32-2498/successes/index.html)

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-07-08/15:31-2529/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-07-08/15:31-2529/successes/index.html)

**Comment by i-like-robots** [comment]
:+1: when :green_heart:

**Comment by next-team** [comment]
I've spotted a [difference](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-07-08/15:46-2531/failures/index.html) between your pull request and production — [see full output](https://s3-eu-west-1.amazonaws.com/ft-next-qa/image_diffs/article/2015-07-08/15:46-2531/successes/index.html)

**Pull request merged**